Modules/Filtering/ImageGrid/test/itkResampleImageFilterGTest.cxx

```diff
@@ -66,6 +66,17 @@ void Expect_ResampleImageFilter_preserves_pixel_value(const TPixel inputPixel)
 
 } // namespace
 
+TEST(ResampleImageFilter, InstantiateTemplateOptions)
+{
+  using PixelType = int;
+  using ImageType = itk::Image<PixelType>;
+
+  const auto filter1 = itk::ResampleImageFilter<ImageType, ImageType >::New();
+  const auto filter2 = itk::ResampleImageFilter<ImageType, ImageType, double, double >::New();
+  const auto filter3 = itk::ResampleImageFilter<ImageType, ImageType, double, float >::New();
+  const auto filter4 = itk::ResampleImageFilter<ImageType, ImageType, float, double >::New();
+  const auto filter5 = itk::ResampleImageFilter<ImageType, ImageType, float, float >::New();
+}
```

Comment:
Instead of creating all those filters at run-time, you could also just instantiate the templates at compile-time, without  using them. For example, at file scope:

    template class itk::ResampleImageFilter<itk::Image<int>, itk::Image<int>, double, float >;
    template class itk::ResampleImageFilter<itk::Image<int>, itk::Image<int>, float , double >;

Then the `TEST(...)` function is not necessary anymore.